application/server/lsp/message_types.go

```diff
@@ -605,24 +606,26 @@ type CodeAction struct {
 	/**
 	 * The workspace edit this code action performs.
 	 */
-	Edit sglsp.WorkspaceEdit `json:"edit,omitempty"`
+	Edit *sglsp.WorkspaceEdit `json:"edit,omitempty"`
```

Comment:
Why pointers?

Reply:
Everywhere I use a pointer is because the field is optional and I want to omit it when it's not set explicitly.

This is important because the way the client knows to call codeAction/resolve is by seeing a CodeAction with no Edit or Command.

I'll edit the comment at the top to make this clearer.

Reply:
TIL